server/player/player.go

```diff
@@ -655,6 +655,21 @@ func (p *Player) FinalDamageFrom(dmg float64, src damage.Source) float64 {
 	return math.Max(dmg, 0)
 }
 
+// Explode ...
+func (p *Player) Explode(c block.ExplosionConfig, impact float64) {
+	p.Hurt(math.Floor((impact*impact+impact)*3.5*c.Size+1), damage.SourceExplosion{})
+
+	vel := p.Position().Sub(c.Pos).Normalize().Mul(impact)
+	res := 0.0
+	for _, i := range p.armour.Items() {
+		if a, ok := i.Item().(item.Armour); ok {
+			res += a.KnockBackResistance()
+		}
+	}
+
+	p.SetVelocity(vel.Mul(1 - res))
```

Comment:
Any reason not to use Player.KnockBack() for this?

Reply:
Player.KnockBack() doesn't work on entities that cannot take damage, whereas explosions allow them.

Reply:
Consider in that case having an unexported `knockBack()` method that does not have that check (or a bool to decide it). That way we can reduce duplicate code.

Reply:
Changed.